File: dev-docs/snippets/prompts/execute.swift
import JsBaoClient

// Execute a prompt by key. Both clients return a typed `ExecutePromptResult`.
func execute(client: JsBaoClient, promptKey: String) async throws {
  // #region example
  let result = try await client.prompts.execute(
    promptKey: promptKey,
    options: ExecutePromptOptions(
      variables: ["topic": "otters", "tone": "playful"],
      modelOverride: "gpt-4o"
    )
  )
  if result.success {
    print(result.output, result.metrics.totalTokens ?? 0)
  }
  // #endregion example
  _ = result
}

File: dev-docs/snippets/prompts/execute.ts
import type { JsBaoClient } from "js-bao-wss-client";

// Execute a prompt by key. JS returns a typed `ExecutePromptResult`.
export async function execute(client: JsBaoClient, promptKey: string) {
  // #region example
  const result = await client.prompts.execute(promptKey, {
    variables: { topic: "otters", tone: "playful" },
    modelOverride: "gpt-4o",
  });
  if (result.success) {
    console.log(result.output, result.metrics.totalTokens);
  }
  // #endregion example
  return result;
}
